software/utils/rpi_config/bootstrap.sh

```diff
@@ -14,81 +14,118 @@
-set -o xtrace
+if [ -n "$VERBOSE" ]; then
+  set -o xtrace
+fi
+
+EXIT_FAILURE=1
+EXIT_SUCCESS=0
 
 # Check if Linux
 PLATFORM="$(uname -s)"
 if [ $PLATFORM != "Linux" ]; then
   echo "Error: This script only supports 'Linux'. You have $PLATFORM."
-  exit 1
+  exit $EXIT_FAILURE
 fi
 
-sudo apt-get --yes install guake git-lfs
-
-### this is for controller debug interface
-pip3 install matplotlib pillow pandas gitpython
-
-### enable serial interface but not console
-sudo raspi-config nonint do_serial 2
+# Make sure we are not in sudo
+if [ "$EUID" -eq 0 ] && [ "$2" != "-f" ]; then
+  echo "Please do not run this script with root privileges!"
+  exit $EXIT_FAILURE
+fi
 
-### disable screen blanking
-sudo raspi-config nonint do_blanking 1
+if [ -z "$VERBOSE" ]; then
+  echo "==============================================================================="
+  echo "=================== RespiraWorks Ventilator bootstrapper ======================"
+  echo "==============================================================================="
+  echo " "
+  echo "  -- clones the repository"
+  echo "  -- configures Raspberry Pi to function as Ventilator GUI:"
+  echo "     * no screensaver or screen blanking"
+  echo "     * GUI<->controller serial interface enabled but no terminal"
+  echo "  -- udev rule added for nucleo, enabling flashing of firmware"
+  echo "  -- guake console accessible at all times using [F12]"
+  echo "  -- desktop shortcuts for ventilator GUI, updater, debug interface"
+  echo "  -- RW-theme desktop background"
+  echo "  -- installs toolchains and dependencies for building controller and GUI"
+  echo "  -- will require a restart when done"
+  echo " "
+  echo " PLEASE DO NOT RUN THIS ON YOUR DEVELOPMENT PC."
+  echo " THIS WILL MESS WITH YOUR BOOTLOADER."
+  echo " THIS IS FOR RASPBERRY-PI ONLY!"
+  read -n1 -s -r -p $'Press any key to continue...\n' key
+fi
 
-### disable splash screen
-sudo raspi-config nonint do_boot_splash 1
-
-### configure USB permissions to deploy to Nucleo
-echo 'ATTRS{idVendor}=="0483", ATTRS{idProduct}=="374b", MODE="666"' | sudo tee /etc/udev/rules.d/99-openocd.rules > /dev/null
+### No screensaver or screen lock
+gsettings set org.mate.power-manager sleep-display-ac 0
+gsettings set org.mate.screensaver lock-enabled 'false'
+gsettings set org.mate.screensaver idle-activation-enabled 'false'
+gsettings set org.mate.screensaver lock-delay 3600
+gsettings set org.mate.screensaver cycle-delay 240
+
+mate-panel --reset --layout netbook
+sudo add-apt-repository --yes ppa:lah7/ubuntu-mate-colours
+sudo apt-get update
+sudo apt-get --yes install ubuntu-mate-colours-blue guake git-lfs
+# \TODO: apply Ambiant-MATE-Dark-Blue theme?
+sudo apt-get --yes remove libreoffice*
+
+### Update the system
+sudo apt-get update
+sudo apt-get --yes upgrade
+sudo apt-get --yes autoremove
+sudo apt-get autoclean
+
+# Enable UART, but disable serial terminal
+echo "enable_uart=1" | sudo tee -a /boot/firmware/usercfg.txt
```

Comment:
This is still an issue, as it turns out that this worked because the `enable_uart=1` in `/boot/firmware/config.txt`is overridden by an `enable_uart=0` in `/boot/firmware/syscfg.txt` which is included later on. If I alter that file to make `enable_uart=1` be the actual setting, the Pi can no longer boot. Though if we don't have it on, the Pi cannot communicate with the STM32...
[This](https://raspberrypi.stackexchange.com/questions/114366/rpi4-serial-port-not-working-on-either-raspberry-os-or-ubuntu) looks promising and I am going to attempt it again soon.

Reply:
Looks to be easier to just configure it manually, which is what is in the new instructions